### ISSUE TITLE
Cohort Sample incorrect initially and correct on refresh

### ISSUE THREAD

**Issue by sarahgbmc** [OHDSI/Atlas#2817]
### Expected behavior
Generating a sample from a cohort results in a list of patients that match the cohort and sample criteria (ie 10 female patients between 25 and 55).

### Actual behavior
When we generate a sample from an existing cohort definition, the initial generation results in patients that do not belong to the cohort and do not meet the sample criteria (for example, if the cohort specifies female patients only, male and female patients appear; choose a sample of 10 patients and >100 are listed). When we refresh the cohort, patients and sample show correctly. 

No errors appear in the log files. Our data is stored in SQL Server. I believe we are running Atlas 2.10.0 (we are not on the most recent version). 

We're not sure if this issue is happening just with our build, or if it could be an Atlas issue. Thank you!
### Steps to reproduce behavior

Create a cohort, generate the cohort, then generate a sample.

Initial generation:
![image](https://user-images.githubusercontent.com/103575490/213193797-48d59fe7-0edc-4553-a8ed-e7399619a3a3.png)

After refresh:
![image](https://user-images.githubusercontent.com/103575490/213193854-07be981a-fce9-4446-b630-7828060e0d60.png)

**Comment by chrisknoll** [comment]
Just for my own information: does it look like the sample it is showing you is coming from a different sample set than you expected?  I'm not sure if you mean 'refresh' as in you refreshed the sample, or you refreshed the webpage (by reloading or closing/reopening).  What steps are involved in 'refresh' that gives you the correct result?

**Comment by chrisknoll** [comment]
I'll never argue with closing an issue :).   Could you let me know which version was the one where it was not working, and which version it was that was? Maybe we can apply a hotfix, or maybe we ask people to upgrade to the specific version.

Thank you!

**Comment by sarahgbmc** [comment]
> I'll never argue with closing an issue :). Could you let me know which version was the one where it was not working, and which version it was that was? Maybe we can apply a hotfix, or maybe we ask people to upgrade to the specific version.
> 
> Thank you!

Yes! We were encountering the issue on Atlas version 2.10.1 (WebAPI version 2.10.1). Issue was resolved when we moved to Atlas version 2.13.0 (same WebAPI version - 2.10.1). I can't say for sure it was an Atlas issue and not an issue with our data sources or active directory set up (or anything else), but we did try to troubleshoot, review log files, etc. Switching versions resolved it immediately. 

Thank you!
